conductor/src/main.rs

```diff
@@ -649,111 +649,37 @@ async fn init_cloud_perms(
     });
 
     // Format Backup spec in CoreDBSpec
-    let backup = Backup {
-        destinationPath: Some(format!(
-            "s3://{}/coredb/{}/org-{}-inst-{}",
-            backup_archive_bucket,
-            &read_msg.message.organization_name,
-            &read_msg.message.organization_name,
-            &read_msg.message.dbname
-        )),
-        encryption: Some(String::from("AES256")),
-        retentionPolicy: Some(String::from("30")),
-        schedule: Some(generate_rand_schedule().await),
-        s3_credentials: Some(S3Credentials {
-            inherit_from_iam_role: Some(true),
+    // If read_msg.message.event_type is Event::Update we want to use the existing backup
+    // configuration spec
+    if read_msg.message.event_type == Event::Update {
+        // Since Backup is always present, there's no need to clone it for an update.
+        // Just return here since we are not updating anything.
+    } else {
+        // Directly assign a new Backup object to coredb_spec.backup for non-Update events
+        coredb_spec.backup = Backup {
+            destinationPath: Some(format!(
+                "s3://{}/coredb/{}/org-{}-inst-{}",
+                backup_archive_bucket,
+                &read_msg.message.organization_name,
+                &read_msg.message.organization_name,
+                &read_msg.message.dbname
+            )),
+            encryption: Some(String::from("AES256")),
+            retentionPolicy: Some(String::from("30")),
+            schedule: Some(generate_rand_schedule()),
```

Comment:
It seems to me like this would reset the schedule to null in the case of an Update, perhaps? It would do an apply of the CoreDB resource type without adding in these configurations?

Reply:
the update is just updating the `CoreDBSpec`.  By not updating the `spec.backup` section during an update it should reuse the same values since we are not overriding or updating anything.  Or that was my thought anyway.